--- newlib-1.11.0/libgloss/mips/crt0.S~	2002-11-12 15:14:24.000000000 -0800
+++ newlib-1.11.0/libgloss/mips/crt0.S	2004-01-02 00:29:27.000000000 -0800
@@ -152,6 +152,7 @@
 	bne	t0,zero,4f
 	nop
 
+	la	a0, __memsize			# a0 is _not_ preserved over calls!
 	/* NOTE: a0[0] contains the amount of memory available, and
 	         not the last memory address. */
 	lw	t0,0(a0)			# last address of memory available
